api/v1/nificluster_types.go

```diff
@@ -46,12 +46,18 @@ type NifiClusterSpec struct {
 	Service ServicePolicy `json:"service,omitempty"`
 	// Pod defines the policy for pods owned by NiFiKop operator.
 	Pod PodPolicy `json:"pod,omitempty"`
+	// clusterManager specifies which manager will handle the cluster election and states.
+	// +kubebuilder:default:=zookeeper
+	// +optional
+	ClusterManager ClusterManagerType `json:"clusterManager,omitempty"`
```

Comment:
We should validate that the `clusterManager` cannot be `kubernetes` unless the NiFi version (pulled from the image tag) is >= 2.0.0*

Reply:
No, and we had no intention of doing so. Because, for example, we can't check the version of NiFi because of a custom image. It's up to the user to use the right functionality with the right version. For example, the operator doesn't work with certain older versions, but nothing prevents the user from using them. The errors present in the operator and in NiFi should be enough to understand that there is a problem.

Reply:
That's fair. I'm not aware of a better way to check the nifi version until the application is already up and running, when it'd be available via the API. So we can consider this convo resolved and route people to the compatibility matrix if they have questions.